lib/iris/experimental/um.py

```diff
@@ -590,12 +590,16 @@ def _update_fixed_length_header(self):
         # Now deal with the LOOKUP and DATA components.
         if self.fields:
             header.lookup_start = word_number
-            lengths = set(field.num_values() for field in self.fields)
-            if len(lengths) != 1:
-                msg = 'Inconsistent header lengths - {}'.format(lengths)
+            header_lengths = [field.num_values() for field in self.fields]
+            if len(set(header_lengths)) != 1:
+                msg = 'Inconsistent header lengths - {}'.format(header_lengths)
```

Comment:
BTW, since iris requires Python 2.7, you could use a set comprehension here instead.